ooniprobe/Model/Database/Measurement.mm

```diff
@@ -96,4 +96,24 @@ -(void)deleteObject{
     [self remove];
 }
 
+- (void)getExplorerUrl:(void (^)(NSString *))measurement_url {
+    NSString *path = [NSString stringWithFormat:@"https://api.ooni.io/api/v1/measurements?report_id=%@&input=%@", self.report_id, self.url_id.url];
+    NSURL *url = [NSURL URLWithString:path];
+    NSURLSessionDataTask *downloadTask = [[NSURLSession sharedSession] dataTaskWithURL:url completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+        if (!error) {
+            NSDictionary *dic = [NSJSONSerialization JSONObjectWithData:data options:0 error:&error];
```

Comment:
Please, after you refactor, make sure you handle an error when loading the JSON.